troubleshoot/troubleshoot.sh

```diff
@@ -0,0 +1,402 @@
+  log_info "checking if connection to cluster is valid ..."
+  curl_params=(
+    -sI
+    -o "/dev/null"
+    "${api_url}/v1/deployment/installer/agent/unix/default/latest/metainfo"
+    "-H" "\"Authorization: Api-Token ${paas_token}\""
+  )
+
+  # proxy
+  proxy=""
+  proxy_secret_name=$("${cli}" get dynakube "${selected_dynakube}" -n "${selected_namespace}" --template="{{.spec.proxy.valueFrom}}")
+  if [[ "$proxy_secret_name" != "" && "$proxy_secret_name" != "$missing_value" ]]; then
+    # get proxy from secret
+    encoded_proxy=$("${cli}" get secret "${proxy_secret_name}" -n "${selected_namespace}" --template="{{.data.proxy}}")
+    proxy=$(echo "$encoded_proxy" | base64 -d)
+    log_info "loading proxy from secret '$proxy_secret_name'"
+  else
+    # try get proxy from dynakube
+    proxyValue=$("${cli}" get dynakube "${selected_dynakube}" -n "${selected_namespace}" --template="{{.spec.proxy.value}}")
+    if [[ "$proxyValue" != "$missing_value" ]]; then
+      proxy=$proxyValue
+    fi
+  fi
+
+  if [[ "$proxy" != "" ]]; then
+    log_info "using proxy: $proxy"
+    curl_params+=("--proxy" "${proxy}")
+  else
+    log_info "proxy is not used"
+  fi
+
+  # skip cert check
+  skip_cert_check=$("${cli}" get dynakube "${selected_dynakube}" -n "${selected_namespace}" --template="{{.spec.skipCertCheck}}")
+  if [[ "$skip_cert_check" == "true" ]]; then
+    log_info "skipping cert check"
+    curl_params+=("--insecure")
+  fi
+
+  # trusted ca
+  custom_ca_map=$("${cli}" get dynakube "${selected_dynakube}" -n "${selected_namespace}" --template="{{.spec.trustedCAs}}")
+  if [[ "$custom_ca_map" != "" && "$custom_ca_map" != "$missing_value" ]]; then
+    # get custom certificate from config map and save to file
+    certs=$("${cli}" get configmap "${custom_ca_map}" -n "${selected_namespace}" --template="{{.data.certs}}")
+    cert_path="/tmp/ca.pem"
+
+    log_info "copying certificate to container ..."
+    ca_cmd="$container_cli \"echo '$certs' > $cert_path\""
+    if ! eval "$ca_cmd"; then
+      error "unable to write custom certificate to container"
+    else
+      log_info "custom certificate successfully written to container!"
+    fi
+
+    log_info "using custom certificate in '$cert_path'"
+    curl_params+=("--cacert" "$cert_path")
+  else
+    log_info "custom certificate is not used"
+  fi
+
+  log_info "trying to access tenant '$api_url' ..."
+  connection_cmd="$container_cli \"curl ${curl_params[*]}\""
+  if ! eval "${connection_cmd}"; then
+    error "unable to connect to tenant"
+  else
+    log_info "tenant is accessible"
+  fi
+}
+
+####### MAIN #######
+
+checkDependencies
+
+checkNs
+checkDynakube
+
+# choose operator pod to check connection/images
+operator_pod=$("${cli}" get pods -n "${selected_namespace}" --no-headers -o custom-columns=":metadata.name" | grep dynatrace-operator)
+log_info "using pod '$operator_pod'"
+container_cli="${cli} exec ${operator_pod} -n ${selected_namespace} -- /bin/bash -c"
```

Comment:
The variable name is misleading and not very descriptive, i think something like this would be better:
```suggestion
run_container_command="${cli} exec ${operator_pod} -n ${selected_namespace} -- /bin/bash -c"
```